### PR TITLE
Implemented maskr

### PR DESCRIPTION
**Pull request by parth121101**
Fix : #3461

### CODE REVIEW

**Review by Pranavchiku — APPROVED** [review]
Looks great, thanks @parth121101 !

**Comment by Pranavchiku** [comment]
An general advice on how your commit history shall be
* implemented maskr function
* add and register test
* update test references

This commits shall suffice, optional commits like:
* applied code review
* marked test as NOFAST, etc.

File: integration_tests/intrinsics_150.f90 [hunk 1]
```diff
@@ -0,0 +1,58 @@
+program intrinsic_147
```
Comment (HarshitaKalani) [inline integration_tests/intrinsics_150.f90:1]:
```suggestion
program intrinsic_150
```

File: src/libasr/pass/intrinsic_functions.h [hunk 2]
```diff
@@ -1792,6 +1793,44 @@ namespace Mod {
 
 } // namespace Mod
 
+namespace Maskr {
+    static ASR::expr_t* eval_Maskr(Allocator& al, const Location& loc,
+            ASR::ttype_t* t1, Vec<ASR::expr_t*>& args, diag::Diagnostics& /*diag*/) {
+        int32_t kind = ASRUtils::extract_kind_from_ttype_t(t1);
+        int64_t i = ASR::down_cast<ASR::IntegerConstant_t>(args[0])->m_n;
+        if(((kind==4) && i>32) || (kind==8 && i>64) || i<0) {
```
Comment (HarshitaKalani) [inline src/libasr/pass/intrinsic_functions.h:1801]:
```suggestion
        if (((kind == 4) && i > 32) || (kind == 8 && i > 64) || i < 0) {
```

File: src/libasr/pass/intrinsic_functions.h [hunk 3]
```diff
@@ -1792,6 +1793,44 @@ namespace Mod {
...
+                return nullptr;
+        }
+        if(i==64){
```
Comment (HarshitaKalani) [inline src/libasr/pass/intrinsic_functions.h:1804]:
```suggestion
        if(i == 64){
```

File: src/libasr/pass/intrinsic_functions.h [hunk 4]
```diff
@@ -1792,6 +1793,44 @@ namespace Mod {
...
+        /*
+        * r = Maskr(x)
+        * r = (1 << x) - 1
+        */
+        ASR::expr_t *sixty_four = i(64, return_type);
+        ASR::expr_t* one = i(1, return_type);
+        ASR::expr_t* minus_one = i(-1, return_type);
+        ASR::expr_t *cast = ASRUtils::EXPR(ASR::make_Cast_t(al, loc, args[0], ASR::cast_kindType::IntegerToInteger, return_type, nullptr));
+        body.push_back(al, b.If((iEq(cast, sixty_four)), {b.Assignment(result,minus_one)}, 
+        {b.Assignment(result, i_tSub(i_BitLshift(one, cast, return_type), one, return_type))}));
```
Comment (HarshitaKalani) [inline src/libasr/pass/intrinsic_functions.h:1827]:
```suggestion
            {b.Assignment(result, i_tSub(i_BitLshift(one, cast, return_type), one, return_type))}));
```

File: integration_tests/intrinsics_150.f90 [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    if(kind(maskr(10,8)) /= 8) error stop
+
+    print *, kind(maskr(10))
+    if(kind(maskr(10)) /= 4) error stop
+
+    print *, kind(maskr(a,4))
+    if(kind(maskr(a,4)) /= 4) error stop
+
+    print *, kind(maskr(a,8))
+    if(kind(maskr(a,8)) /= 8) error stop
```
Comment (HarshitaKalani) [inline integration_tests/intrinsics_150.f90:55]:
```suggestion
    if (kind(maskr(a,8)) /= 8) error stop
```
Similarly in the above lines.

**Review by HarshitaKalani — COMMENTED** [review]
I left a few comments. Rest looks good. Thanks!

**Review by certik — APPROVED** [review]
Thanks!

**Comment by certik** [comment]
Please resolve conflicts, then we can merge it.

**Pull request merged**